index.d.ts

```diff
@@ -0,0 +1,73 @@
+import { ITypedField, ITypedValue, ITypedData } from 'eth-sig-util';
+import { Transaction } from 'ethereumjs-tx';
+export { ITypedField, ITypedValue, ITypedData }
+
+type WalletOpts = {
+  withAppKeyOrigin?: string
+}
+
+export interface IKeyring {
+  serialize(): Promise<any>
+  deserialize(serialized:any): Promise<void>
+
+  addAccounts(numberToAdd: number): Array<string> // returns address array
+  getAccounts(): Promise<Array<string>>
+  removeAccount(address:string): Promise<void>
```

Comment:
`eth-hd-keyring` seems to be missing this method. And in `eth-simple-keyring` it returns `void` (e.g. it isn't async)